### DISCUSSION TITLE
Specifiche tecniche

### DISCUSSION THREAD

**Discussion by rossoc** [Project-SWEnergy/doc-latex#173] [General]
Discussione sul documento specifiche tecniche

**Comment by rossoc** [comment]
In particolare, è stata scritta la prima bozza delle sezioni:
- [x] Introduzione
- [x] Tecnologie adottate
- [x] Architettura di deployment
- [ ] Architettura implementativa
- [x] Requisiti soddisfatti

---

## Architettura di deployment
- Trattata solo l'architettura monolitica del backend. Il dubbio è: il frontend adotta un'architettura differente?
## Tecnologie adottate
Mancano:
- test in angular
- test in nest
- tecnologie della chat
- citare html
## Architettura Implementativa
- non è stato scritto niente
- potremmo scrivere inerentemente alla dependency injection
- aggiungiamo i pattern mano a mano che li usiamo

## Requisiti soddisfatti 
- mancano le tabelle dei grafici a torta che mostrino rispettivamente
    - requisiti obbligatori soddisfatti e non
    - requisiti obbligatori/facoltativi/desiderabili soddisfatti e non

**Comment by alvide** [comment]
Le correzioni che ho fatto su questo documento sono solo relative a delle sviste/refusi, non è stata fatta nessuna modifica di natura contenutistica.

Per quanto riguarda l'Architettura di deployment lato frontend dovrebbe essere anche essa monolitica.
Per quanto riguarda i test in angular Giacomo aveva visto/approfondito dei test con Jasmine e Karma che permettevano di fare unit test. (chiedere a Giacomo per maggiori dettagli).
